sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/PartitionProcessorImpl.java

```diff
@@ -175,8 +189,13 @@ public PartitionProcessorImpl(ChangeFeedObserver observer, ChangeFeedContextClie
 
                 return true;
             })
-            .onErrorResume(throwable -> Flux.empty())
-            .then();
+            .onErrorResume(throwable -> {
+                if (this.resultException == null) {
+                    this.resultException = new RuntimeException(throwable);
```

Comment:
just please be aware that this doesn't record exception stacktrace. If that matters. this will maintain the original exception stacktrace though.

Is that ok?

Reply:
just like above, yes, the reason we create a RuntimeException here and assign it to the resultException is to signal to the "parent" that the processing worker should be killed along with the corresponding renewal worker.